### PR TITLE
feat: reply component

### PR DESCRIPTION
**Pull request by sugarpusher**

### CODE REVIEW

**Review by sugarpusher — COMMENTED** [review]
I created this component as a <li> because it seems to be the semantically correct element, however if a <div> would be more appropriate, let me know and I will change it.

File: src/reply/reply.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+export type ReplyProps = h.JSX.HTMLAttributes<HTMLLIElement> & {
+	replyTitle?: string
+	replyDate?: string
+	replierName?: string
+	replierRep?: number
+	replyAmt?: number
+	myReply?: boolean
+	avatar?: string
+	detail?: boolean
```
Comment (vojtechsimetka) [inline src/reply/reply.tsx:14]: Shouldn't some of these be required?

File: src/reply/reply.tsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
+import { h } from 'preact'
+import cn from 'classnames'
+import classes from './reply.module.css'
+import defaultAvatar from './defaultAvatar.jpg'
+
+export type ReplyProps = h.JSX.HTMLAttributes<HTMLLIElement> & {
+	replyTitle?: string
+	replyDate?: string
```
Comment (vojtechsimetka) [inline src/reply/reply.tsx:8]: IMHO this should be of type `Date` and we format it somewhow.

File: src/reply/reply.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+		replyTitle: {
+			name: 'Reply Title',
+			type: { name: 'string', required: true },
+			defaultValue: 'Hi Sally, I can do it for you!',
+		},
+		replyDate: {
+			name: 'Reply Date',
+			type: { name: 'string', required: true },
+			defaultValue: 'Today, 15:01',
+		},
```
Comment (vojtechsimetka) [inline src/reply/reply.stories.tsx:18]: The type should be `date`

File: src/reply/reply.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+		replierRep: {
+			name: 'Replier Reputation',
+			type: { name: 'number', required: true },
+			defaultValue: 52,
+		},
+		replyAmt: {
+			name: 'Reply Amount',
+			type: { name: 'number', required: true },
+			defaultValue: 12,
+		},
```
Comment (vojtechsimetka) [inline src/reply/reply.stories.tsx:33]: Lets keep it as `number` now but we probably need to change the type to `BigNumber` or `bigint` depending on the implementation we choose.

**Comment by vojtechsimetka** [comment]
TODO from my side: figure out how to format the `Date`

File: src/reply/reply.stories.tsx (hunk 4 above)
Comment (filoozom) [reply src/reply/reply.stories.tsx:34]: I'd probably tend towards `bigint` for futureproofing, even though it likely makes the bundle a bit larger right now.

**Review by vojtechsimetka — APPROVED** [review]

**Pull request merged**